source/common/buffer/buffer_impl.cc

```diff
@@ -40,6 +40,12 @@ void OwnedImpl::addDrainTracker(std::function<void()> drain_tracker) {
   slices_.back().addDrainTracker(std::move(drain_tracker));
 }
 
+std::weak_ptr<Account> OwnedImpl::account() const { return account_; }
+void OwnedImpl::bindAccount(std::weak_ptr<Account> account) {
```

Comment:
Also verify that the buffer has no slices when associating the buffer with an account.

`ASSERT(slices_.empty());`